partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/ConfigParser.scala

```diff
@@ -0,0 +1,41 @@
+package uk.gov.ukho.ais.partitioning
+
+import scopt.OParser
+
+case class Config(inputPath: String, outputDirectory: String) {}
+
+object ConfigParser {
+  @transient private final val PARSER = {
+    val builder = OParser.builder[Config]
+
+    import builder._
+    OParser.sequence(
+      programName("partition-raw-ais"),
+      head("Partition Raw AIS",
+           "Partitions the raw AIS data by month and year"),
+      help('h', "help"),
+      opt[String]('i', "input")
+        .required()
+        .valueName("<input data path/S3 URI>")
+        .text("path of input data")
+        .action((value, config) => config.copy(inputPath = value)),
+      opt[String]('o', "output")
+        .required()
+        .valueName("<output directory/S3 bucket>")
+        .text("location to output the resulting partitioned files")
+        .action((value, config) => config.copy(outputDirectory = value))
+    )
+  }
+
+  def parse(args: Array[String]): Config =
+    OParser
+      .parse(
+        PARSER,
+        args,
+        Config(
+          inputPath = "",
+          outputDirectory = ""
+        )
+      )
+      .fold { throw new IllegalStateException("config not loaded") }(identity)
```

Comment:
Isn't ```_``` the identity function?

```suggestion
      .fold { throw new IllegalStateException("config not loaded") }(_)
```

Reply:
Doesn't work for some reason.

Reply:
Fair enough

Reply:
identity is legit scala as far as I can see.